crates/turborepo-lib/src/run/global_hash.rs

```diff
@@ -125,11 +142,15 @@ pub fn get_global_hash_inputs<'a, L: ?Sized + Lockfile>(
 impl<'a> GlobalHashableInputs<'a> {
     pub fn calculate_global_hash_from_inputs(mut self) -> String {
         match self.env_mode {
-            EnvMode::Infer if !self.pass_through_env.is_empty() => {
+            // In infer mode, if there is any pass_through config (even if it is an empty array)
+            // we'll hash the whole object, so we can detect changes to that config
+            // Further, resolve the envMode to the concrete value.
+            EnvMode::Infer if self.pass_through_env.is_some() => {
```

Comment:
Classic `nil` is not empty bug

Reply:
IIUC, there is a distinction between empty array and not set?

Reply:
Yep, in this case due to our code but also capnproto makes that distinction too